src/main/java/org/ibp/api/java/impl/middleware/inventory/manager/LotServiceImpl.java

```diff
@@ -217,4 +232,54 @@ public void mergeLots(final String keepLotUUID, final LotsSearchDto lotsSearchDt
 		this.lotService.mergeLots(loggedInUser.getUserid(), lotDto.getLotId(), lotsSearchDto);
 	}
 
+	@Override
+	public void splitLot(final String programUUID, final LotSplitRequestDto lotSplitRequestDto) {
+
+		this.lotSplitValidator.validateRequest(lotSplitRequestDto);
+
+		List<String> lotUUIDs = Arrays.asList(lotSplitRequestDto.getSplitLotUUID());
+		final LotsSearchDto lotsSearchDto = new LotsSearchDto();
+		lotsSearchDto.setLotUUIDs(lotUUIDs);
+		final List<ExtendedLotDto> extendedLotDtos = this.searchLots(lotsSearchDto, null);
+		this.extendedLotListValidator.validateAllProvidedLotUUIDsExist(extendedLotDtos, new HashSet<>(lotUUIDs));
+
+		ExtendedLotDto splitLotDto = extendedLotDtos.get(0);
+		LotSplitRequestDto.InitialLotDepositDto initialDeposit = lotSplitRequestDto.getInitialDeposit();
+		this.lotSplitValidator.validateSplitLot(splitLotDto, initialDeposit.getAmount());
+
+		//Creates the new lot
+		final LotSplitRequestDto.NewLotSplitDto newLot = lotSplitRequestDto.getNewLot();
+		final LotGeneratorInputDto lotGeneratorInputDto = new LotGeneratorInputDto();
+		lotGeneratorInputDto.setGid(splitLotDto.getGid());
+		lotGeneratorInputDto.setUnitId(splitLotDto.getUnitId());
+		lotGeneratorInputDto.setLocationId(newLot.getLocationId());
+		lotGeneratorInputDto.setNotes(newLot.getNotes());
+		lotGeneratorInputDto.setGenerateStock(newLot.getGenerateStock());
+		lotGeneratorInputDto.setStockPrefix(newLot.getStockPrefix());
+		String newLotUUID = this.saveLot(programUUID, lotGeneratorInputDto);
```

Comment:
Instead of calling this.saveLot I will prefer to use lotService.saveLot. Extracting the stockId generation to a new reusable function. This will allow the validations to run first of any save in the database and avoid calling unnecessary validations like unit Id